src/pykeen/nn/representation.py

```diff
@@ -1532,8 +1559,128 @@ def _plain_forward(self, indices: LongTensor | None = None) -> FloatTensor:  # n
+    kwargs: OptionalKwargs | None = None
+
+    def __post_init__(self) -> None:
+        """Implement data integrity checks."""
+        if len(set(self.ids)) != len(self.ids):
+            raise ValueError(f"Duplicate in {self.ids=}")
+        if any(i < 0 for i in self.ids):
+            raise ValueError(f"Negative ids in {self.ids=}")
+
+    def get_base(self) -> Representation:
+        """Instantiate the base representation."""
+        # import here to avoid cyclic import
+        from . import representation_resolver
+
+        max_id = len(self.ids)
+
+        base = representation_resolver.make(self.base, self.kwargs, max_id=max_id)
+        if base.max_id != max_id:
+            raise MaxIDMismatchError(
+                f"When constructing the backfill specification, got a mismatch between the number of IDs "
+                f"given ({max_id:,}) and the max_id assigned to the base representation ({base.max_id:,})"
+            )
+
+        return base
+
+
+@parse_docdata
+class MultiBackfillRepresentation(PartitionRepresentation):
+    """Fill missing ids by backfill representation.
+
+    ---
+    name: Multi-Backfill
+    """
+
+    def __init__(
+        self,
+        *,
+        max_id: int,
+        partitions: Sequence[Partition],
+        shape: OneOrSequence[int] | None = None,
+        backfill: HintOrType[Representation] = None,
+        backfill_kwargs: OptionalKwargs = None,
+        **kwargs,
+    ) -> None:
+        """Initialize the backfill representation."""
+        # import here to avoid cyclic import
+        from . import representation_resolver
+
+        bases: list[Representation] = []
+        # format: (base_index, local_index)
+        assignment = torch.zeros(size=(max_id, 2), dtype=torch.long)
+        backfill_mask = torch.ones(assignment.shape[0], dtype=torch.bool)
+        all_ids: set[int] = set()
+        for base_index, partition in enumerate(partitions):
+            # check for overlap with others
+            if colliding_ids := all_ids.intersection(partition.ids):
+                raise ValueError(f"{colliding_ids=} for bases[{base_index}] with {partition.ids=}")
+            all_ids.update(partition.ids)
+
+            # set assignment
+            ids_t = torch.as_tensor(partition.ids, dtype=torch.long)
+            assignment[ids_t, 0] = base_index
+            assignment[ids_t, 1] = torch.arange(len(partition.ids))
+            backfill_mask[ids_t] = False
+
+            bases.append(partition.get_base())
+
+        # shape verification
+        shapes = [base.shape for base in bases]
+        if len(set(shapes)) != 1:
+            raise ShapeMismatchError(f"Base instances had multiple different shapes: {shapes}")
+        if shape is None:
+            shape = shapes[0]
+        elif shapes[0] != shape:
+            raise ShapeMismatchError(
+                f"The explicitly given {shape=} was different than the shape of the bases {shapes[0]}"
+            )
```

Comment:
I think this is a duplication of existing functionality in `pykeen.nn.utils.ShapeError`:

```suggestion
        shape = ShapeError.verify(shapes[0], shape)
```